### DISCUSSION TITLE
What is the best way to bring colors from the backend

### DISCUSSION THREAD

**Discussion by LucasLuisBorges** [tailwindlabs/tailwindcss#11455] [Help]
I'm wanting to make a color theme that is dynamic since some colors will come from the backend like background and text. What would be the best way to pass these colors in the tailwind theme.

I created a ::root in global.css:
:root {
  --dashboard-color-100: #304D63;
  --dashboard-color-200: #10202B;
  --visuals-text-color: #fff;
  --accent-color: #fff;
}

And this is my tailwind.config.js:
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ['src/**/*.{ts,tsx}'],
  theme: {
    extend: {
      colors: {
        primary: "#304D63",
        secondary: "#10202B",
        white: "#FFF",
        "dashboard": {
          100:  "var(--dashboard-color-100)",
          200: "var(--dashboard-color-200)",
        },
        "text": {
          100: "var(--visuals-text-color)",
        },
        "accent": {
          100: "var(--accent-color)",
        }
      }
    },
    screens: {
      ssm: '440px',
      sm: '640px',
      md: '768px',
      lg: '1024px',
      xl: '1280px',
      '2xl': '1536px',
    },
  },
  plugins: [],
};

**Comment by wongjn** [comment]
What you currently have should work.

**Reply by LucasLuisBorges** [reply]
Yes, but is it a good way to use it?

**Reply by wongjn** [reply]
I'd say it is good. For the best, consider setting the variables as HSL or RGB component values so that [you can use opacity modifiers with them](https://tailwindcss.com/docs/customizing-colors#using-css-variables).

**Comment by adarshaacharya** [comment]
btw how do you load color from backend and pass it over to the variable.

**Reply by wongjn** [reply]
You could render a `<style>` element that has a rule with the CSS variable value definitions.

**Reply by adarshaacharya** [reply]
not sure if we can render the style tag in between jsx / html?

**Reply by wongjn** [reply]
Sure you can!

```jsx
const Foo = () => {
  const color = backendCall();
  return (
    <style>{`
      :root {
        --my-color: ${color};
      }
    `}</style>
  );
};
```
